core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java

```diff
@@ -1067,20 +1067,22 @@ public String getModuleCallables(String packageName,
             return null;
         }
 
+        GsonBuilder builder = new GsonBuilder();
+        Gson gson = builder.create();
+
         // Go through each callable, parse fasten uri, insert signature.
         for (Object j : json) {
-            JSONObject jObj = (JSONObject) j;
-            var uri = jObj.getString("fasten_uri");
-
             try {
-                var uriObject = FastenUriUtils.parsePartialFastenUri(uri);
-                jObj.put("method_name", uriObject.get(2));
-                jObj.put("method_args", uriObject.get(3));
+                JSONObject jObj = (JSONObject) j;
+                var partialUri = jObj.getString("fasten_uri");
+                var fullUri = FastenUriUtils.generateFullFastenUri("mvn", packageName, packageVersion, partialUri);
+                var uriObj = new FastenJavaURI(fullUri);
+                var js =  gson.toJson(uriObj);
```

Comment:
Although we discussed it multiple times that the implementation of API must not depend on Java, it currently does. In this example, a higher level of refactoring needs to be done for it to be independent (e.g., pass forge along from API, etc). I suggest leaving it like this for now. I will come back to refactoring API as a whole later.